lib/plugins/aws/deploy/lib/uploadArtifacts.js

```diff
@@ -18,7 +18,18 @@ module.exports = {
       Body: body,
       ContentType: 'application/json',
     };
-
+    if (this.serverless.service.provider.deploymentBucketOptions) {
```

Comment:
Curious if it is necessary to have camelCase properties that then have to be mapped to their AWS service equivalent representation? i.e. sse to ServerSideEncryption.

I'm not familiar with standards usually applied here, but it just occurred to me after looking at how the resource configuration works - https://serverless.com/framework/docs/providers/aws/guide/resources/

If the serverless.yml config looked like:

```
deploymentBucketOptions:
  ServerSideEncryption: aws:kms
  SSEKMSKeyId: 123
  SSECustomerKey: 321
```
This block would become
```
let params = {
    Bucket: this.bucketName,
    Key: `${this.serverless.service.package.artifactDirectoryName}/${compiledTemplateFileName}`,
    Body: body,
    ContentType: 'application/json',
};
if (this.serverless.service.provider.deploymentBucketOptions) {
  params = Object.assign(params, this.serverless.service.provider.deploymentBucketOptions);
}
```
Could be simpler, but at the very least you could avoid duplicating that code by creating a method that can apply the sse options.